lib/create-plugin.js

```diff
@@ -570,4 +602,60 @@ function updateOperationSegmentName(context, operationSegment) {
   return false
 }
 
+/**
+ * Captures both field and args of resolvers as metrics.
+ *
+ * This is intended to be used to determine if a field within a graphql schema is still being requested.
+ *
+ * @param {Object} params
+ * @param {Object} params.config plugin config
+ * @param {Object} params.info info key from resolver context
+ * @param {Object} params.operationSegment operation segment
+ * @param {Object} params.args args key from resolver context
+ *
+ */
+function maybeCaptureFieldMetrics({ config, info, operationSegment, args }) {
+  if (config.captureFieldMetrics) {
+    const transaction = operationSegment.transaction
+    const fieldName = info.fieldName
+    const fieldType = info.parentType.toString()
+    captureFieldMetrics({ transaction, args, fieldName, fieldType })
+  }
+}
+
+/**
+ * Used to create metrics that just increment call count.  Intended to be
+ * used to report on when we see args in a resolver
+ *
+ * @param {Object} agent active agent instance
+ * @param {String} name metric name
+ */
+function createCallCountMetric(transaction, name) {
+  const metric = transaction.metrics.getOrCreateMetric(name)
+  metric.incrementCallCount()
+}
+
+/**
+ * Captures both field and args of resolvers as metrics.
+ *
+ * This is intended to be used to determine if a field within a graphql schema is still being requested.
+ *
+ * @param {Object} params
+ * @param {Object} params.transaction active transaction
+ * @param {Object} params.args args key from resolver context
+ * @param {Object} params.fieldType parent type of field
+ * @param {Object} params.fieldName name of field
+ *
+ */
+function captureFieldMetrics({ transaction, args, fieldType, fieldName }) {
+  const fieldMetric = `${FIELD_PREFIX}/${fieldType}.${fieldName}`
+  createCallCountMetric(transaction, fieldMetric)
+  Object.entries(args).forEach(([key]) => {
+    if (fieldName && fieldType) {
```

Comment:
What is this check for? What would we record for `fieldMetric` above if either of these is falsey?

Reply:
shit, this was applicable before i moved things around